src/formats/tana/tana-import.ts

```diff
@@ -0,0 +1,306 @@
+import { TanaDatabase, TanaDoc } from './models/tana-json';
+
+const inlineRefRegex = /<span data-inlineref-node="(.+)"><\/span>/g;
+const boldRegex = /<b>(.*?)<\/b>/g;
+const italicRegex = /<i>(.*?)<\/i>/g;
+const strikeRegex = /<strike>(.*?)<\/strike>/g;
+const codeRegex = /<code>(.*?)<\/code>/g;
+
+export class TanaGraphImporter {
+	public result: Map<string, string> = new Map();
+	private tanaDatabase: TanaDatabase;
+	private nodes: Map<string, TanaDoc>;
+	private convertedNodes: Set<string> = new Set();
+	public fatalError: string | null;
+	public notices: string[] = [];
+	private anchors: Set<string> = new Set();
+	private topLevelNodes = new Map<string, [TanaDoc, string]>();
+
+	public importTanaGraph(data: string) {
+		this.tanaDatabase = JSON.parse(data) as TanaDatabase;
+		this.nodes = new Map();
+		this.tanaDatabase.docs.forEach(n => this.nodes.set(n.id, n));
+
+
+		const rootNode = this.tanaDatabase.docs.find(n => n.props.name && n.props.name.startsWith('Root node for'));
+		if (!rootNode) {
+			this.fatalError = 'Root node not found';
+			return;
+		}
+		this.convertedNodes.add(rootNode.id);
+
+		this.prepareAnchors(rootNode);
+
+		const workspaceNode = this.nodes.get(rootNode.children[0]);
+		if (!workspaceNode) {
+			this.fatalError = 'Workspace node not found';
+			return;
+		}
+		this.convertedNodes.add(workspaceNode.id);
+		this.topLevelNodes.set(workspaceNode.id, [workspaceNode, workspaceNode.props.name]);
+
+		let metaNodeId = workspaceNode.props._metaNodeId;
+		if (metaNodeId) {
+			const metaNode = this.nodes.get(metaNodeId);
+			if (metaNode) {
+				this.markSeen(metaNode);
+			}
+		}
+
+		const libraryNode = this.nodes.get(rootNode.id + '_STASH');
+		if (libraryNode != null) {
+			this.importLibraryNode(libraryNode);
+		}
+		else {
+			this.notices.push('Library node not found');
+		}
+
+		for (let suffix of ['_TRASH', '_SCHEMA', '_SIDEBAR_AREAS', '_USERS', '_SEARCHES', '_MOVETO', '_WORKSPACE', '_QUICK_ADD']) {
+			const specialNode = this.nodes.get(rootNode.id + suffix);
+			if (specialNode != null) {
+				this.markSeen(specialNode);
+			}
+			else {
+				this.notices.push('Special node ' + suffix + ' not found');
+			}
+		}
+
+		this.enumerateChildren(workspaceNode, (childNode) => {
+			if (childNode.props._docType == 'journal') {
+				this.importDailyNotes(childNode);
+			}
+		});
+
+		for (const [node, file] of this.topLevelNodes.values()) {
+			this.convertNode(node, file);
+		}
+
+		this.notices.push('Converted ' + this.convertedNodes.size + ' nodes');
+		let unconverted = 0;
+		for (let node of this.tanaDatabase.docs) {
+			if (!this.convertedNodes.has(node.id) && !node.id.startsWith('SYS') &&
+				node.props._docType != 'workspace') {
+				const path = this.pathFromRoot(node, rootNode);
+				if (path) {
+					this.notices.push('Found unconverted node: ' + path);
+					unconverted++;
+					if (unconverted == 50) break;
+				}
+			}
+		}
```

Comment:
For clarity, I would suggest clearing the class-level variables which are only relevant to this call to `importTanaGraph`. However, I think it would be even better to adjust this function to instead take the `ProgressReporter` and to convert and save each file to the vault and report progress as it goes along and to not leavea any state behind on the class after being called. Take a look at some of the other format importers to see this pattern.